events_structs.go

```diff
@@ -855,3 +858,69 @@ func (ev *EventStackChangedType) String() string {
 	d := ev.TypeData
 	return fmt.Sprintf("Stack changed from '%s' to %s", d.PreviousStackName, d.CurrentStackName)
 }
+
+// Database maintenance planned
+type EventDatabaseMaintenancePlannedTypeData struct {
+	AddonName     string `json:"addon_name"`
+	MaintenanceID string `json:"maintenance_id"`
+}
+
+type EventDatabaseMaintenancePlannedType struct {
+	Event
+	TypeData EventDatabaseMaintenancePlannedTypeData `json:"type_data"`
+}
+
+func (ev *EventDatabaseMaintenancePlannedType) String() string {
+	return fmt.Sprintf("A maintenance has been scheduled on the %s addon (Maintenance ID: %s).", ev.TypeData.AddonName, ev.TypeData.MaintenanceID)
+}
+
+func (ev *EventDatabaseMaintenancePlannedType) Who() string {
+	if ev.TypeData.AddonName != "" {
+		return fmt.Sprintf("Addon %s", ev.TypeData.AddonName)
+	}
+	return ev.Event.Who()
+}
```

Comment:
From slack discussion I talked more about the following format:
```go
A maintenance (ID 65004006a8acb50076cf9c1b) has been scheduled on the Redis addon. <scalingo-platform (deploy@scalingo.com)>
```
Two things:
- **non-blocking**: in order to make as small as possible the output, move the "Maintenance ID" right after "maintenance" using only `(ID: 123)`
- **blocking**: The user is missing here

Reply:
I updated the message format, but the user is well added here. Just not handled in this file.